### DISCUSSION TITLE
Should we release 0.72.7?

### DISCUSSION THREAD

**Discussion by huntie** [reactwg/react-native-releases#91] [Patches]
## Released, please refer to #93

# Should we release 0.72.7?

Current Release: [0.72.6](https://github.com/facebook/react-native/releases/tag/v0.72.6)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of qualified picks

1. https://github.com/facebook/react-native/commit/ae0d714bbdd2eb9d114c17c3945b2f678f60b65a - picked
2. https://github.com/facebook/react-native/commit/265af222aa8acb3b514e22a7db191d66755f553b - picked
3. https://github.com/facebook/react-native/commit/18c9797ecc8dab220ec68aae011ddb73db17a99d - https://github.com/facebook/react-native/pull/41377
4. https://github.com/facebook/react-native/commit/10c5192c1d154e8fb9dfff870d02efe5a3b8e672 - picked
6. https://github.com/facebook/react-native/commit/4eed12b7df5d9731e556b77d701c87dcf91d3a1f - https://github.com/facebook/react-native/pull/41378
7. https://github.com/facebook/react-native/commit/daedbe6e3e4cb4b5fde533bf8205edd89a18c480 - https://github.com/facebook/react-native/pull/41376

#### Local commits to backport to main

1.

**Comment by stianjensen** [comment]
Regression with rotation in animated values in RN0.72:
https://github.com/facebook/react-native/commit/ae0d714bbdd2eb9d114c17c3945b2f678f60b65a
(explained here: https://github.com/facebook/react-native/issues/36608)

**Reply by fortmarek** [reply]
Picked ✅

**Comment by malikzype** [comment]
On RN 72.5 and 0.72.6 with Xcode 15, we are facing an issue:
<img width="1240" alt="Screenshot 2023-10-13 at 11 30 03 AM" src="https://github.com/reactwg/react-native-releases/assets/113113102/503b6e0e-f260-4a18-8abe-f7fa0c99300c">

Similar Issues which are open yet -
 https://github.com/facebook/react-native/issues/38283
https://github.com/facebook/react-native/issues/38140
https://github.com/react-native-tvos/react-native-tvos/issues/575

Note: The issue is not replicating on 0.72.4 with XCode 15
Can we release a patch to fix this?

**Reply by cipolleschi** [reply]
Can you create a repro with all the libraries required?
Vanilla react-native apps do not suffer from that problem, but to fix it for everyone, I'll need a reproducer I can work with.

**Reply by malikzype** [reply]
Sure

**Reply by cipolleschi** [reply]
I found that one of the libraries causing this is `react-native-gesture-handler`. We have a [PR to fix that](https://github.com/software-mansion/react-native-gesture-handler/pull/2635).

RNGestureHandler may not be the only one, though.

**Reply by malikzype** [reply]
Great. I can help with a reproducer in a couple of days and shall post here.

**Reply by malikzype** [reply]
Just to confirm that even updating RN Gesture Handler, the issue remains the same.

**Comment by cipolleschi** [comment]
Please pick: 
* https://github.com/facebook/react-native/pull/41009 this stabilizes the React-RCTAppDelegate.podspec

**Reply by fortmarek** [reply]
Picked ✅

**Comment by majkpal** [comment]
Please cherry-pick this fix: 
* [facebook/react-native@265af22](https://github.com/facebook/react-native/commit/265af222aa8acb3b514e22a7db191d66755f553b)


At Chime X Software Mansion we need this fix to be cherry-picked as currently there is no option to upgrade to the next version ( this fix has not been released anyway) and we are blocked with our next project there.

**Reply by fortmarek** [reply]
Picked ✅

**Comment by Mugilan-Codes** [comment]
what version of ruby should be used in version `0.72.6`?

**Reply by liamjones** [reply]
The Gemfile specifies/enforces a range: https://github.com/facebook/react-native/blob/v0.72.6/packages/react-native/template/Gemfile#L4

**Comment by cipolleschi** [comment]
Please pick:
- https://github.com/facebook/react-native/commit/daedbe6e3e4cb4b5fde533bf8205edd89a18c480
- https://github.com/facebook/react-native/commit/4eed12b7df5d9731e556b77d701c87dcf91d3a1f

**Reply by fortmarek** [reply]
Both cause merge conflicts, created separate PRs:
- https://github.com/facebook/react-native/pull/41376
- https://github.com/facebook/react-native/pull/41378

**Comment by cipolleschi** [comment]
merge:  https://github.com/facebook/react-native/pull/41276 or you can't release as CI won't be green.

**Reply by fortmarek** [reply]
Merged ✅

**Comment by vksgautam1986** [comment]
can we have this https://github.com/facebook/react-native/pull/40914

**Reply by stianjensen** [reply]
Link to merged commit on main for that:

https://github.com/facebook/react-native/commit/18c9797ecc8dab220ec68aae011ddb73db17a99d

**Reply by fortmarek** [reply]
This PR causes merge conflict, created a PR: https://github.com/facebook/react-native/pull/41377

**Comment by szymonrybczak** [comment]
CLI Bump: https://github.com/facebook/react-native/pull/41316

**Reply by fortmarek** [reply]
Merged ✅

**Comment by wfern** [comment]
Can we pick this: https://github.com/facebook/react-native/commit/ab81c16b6507000139d1af52c622e320629750ec?

**Reply by fortmarek** [reply]
The `InspectProxy.js` file was part of the `metro-inspect-proxy` package in the Metro repo and it only got moved after 0.72 release: https://github.com/facebook/react-native/pull/39045

I don't think there's an easy way to pick this, so you might need to wait for the 0.73 release cc @huntie

**Reply by wfern** [reply]
Thanks for the info.

Yes, I think wait for 0.73 it's the best option too, I tried the new chrome debugger in the RC and it's amazing!

If I apply a patch from the commit I requested to `metro-inspect-proxy` it fixes the issue because I wasn't even able to see the hermes target in newer versions of chrome/edge. Even so, I can't debug properly on version 0.72 due to other errors that appear in the console, for example:
```
WARN Not connected to page: 29 , failed trying to handle event: ...
```
​

**Comment by liamjones** [comment]
It was suggested that the webp vulnerability patched in Fresco should probably be backported to supported versions of RN - just adding a note here for reference: https://github.com/react-native-community/discussions-and-proposals/issues/725#issuecomment-1780222618

**Comment by tido64** [comment]
I have a couple of fixes for pnpm repos that would be great to include. They've landed on `main`, but required some minor adjustments for `0.72-stable`:
- https://github.com/facebook/react-native/pull/41445
- https://github.com/facebook/react-native/pull/41456
